### DISCUSSION TITLE
Using texture objects in PyCUDA

### DISCUSSION THREAD

**Discussion by GR-DARES** [inducer/pycuda#419] [Q&A] [answered]
Hello everyone!

Recent CUDA versions (starting at CUDA 11.3 if I'm not wrong) deprecated texture references. I have been running CUDA 11.2 and PyCUDA for a while and have several working kernels using texture references.

Recently I upgraded to CUDA 12.2.0 and I'm getting an error when compiling an existing kernel that was using texture references. The solution is to refactor the texture reference into a texture object. I've been looking into the PyCUDA documentation but haven't been able to find how to build a texture object (https://documen.tician.de/pycuda/driver.html#pycuda.driver.TextureReference).

Is there any way to build a texture object and pass it to the kernel call in PyCUDA?

Thanks!

Gerard

**Comment by inducer** [comment, marked as answer]
That functionality isn't currently wrapped. Contributions are welcome!

**Reply by GR-DARES** [reply]
Thanks! I am considering contributing but I'm afraid I won't be able to do it until September.

**Comment by inducer** [comment]
Looking into a bit, this should be mostly a straightforward wrapping exercise. Texture objects are simply passed to kernels by value, so giving them the buffer interface should allow the struct packer to do the right thing automatically.

https://github.com/NVIDIA/cuda-samples/blob/5f97d7d0dff880bc6567faa4c5e62e389a6d6999/Samples/0_Introduction/simpleTextureDrv/simpleTextureDrv.cpp#L222-L223
